src/System.Text.Json/docs/SerializerProgrammingModel.md

```diff
@@ -86,475 +78,144 @@ namespace System.Text.Json.Serialization
-{
-    public sealed class JsonClassInfo
-    {
-        public System.Type Type { get; }
+If the internal converter is not sufficient, such as when the format has a custom format or is not ISO 8601 compatible, then a developer will be able to add a new value converter.
 
-        public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName);
-        public JsonPropertyInfo GetProperty(string propertyName);
+## Enum Supprt
+By default, Enums are treated as longs in the JSON. This is most efficient. There will be future support to be based on strings; likely through an attribute to change the default for a given property and an option on `JsonSerializerOptions` to change the default globally.
 
-        public void AddDataTypeConverter(JsonDataTypeConverterAttribute attribute);
-        public JsonPropertyValueAttribute PropertyPolicyDefault { get; set; }
-        public JsonPropertyNamePolicyAttribute PropertyNamePolicyDefault { get; set; }
-    }
+## ICollection and Array Converter feature
+By default there is an internal converter that supports any concrete class that implements IList<T> and an Array converter that supports jagged arrays (`foo[][]`) but not multidimensional (`foo[,]`).
 
-    public abstract class JsonPropertyInfo
-    {
-        public string EscapedName { get; }
-        public string Name { get; }
-        public PropertyInfo PropertyInfo { get; }
+Work is underway to support collections in the System.Collections.Immutable namespace and others automatically.
 
-        public JsonDataTypeConverterAttribute DataTypeConverter { get; set; }
-        public JsonPropertyValueAttribute PropertyPolicyDefault { get; set; }
-        public JsonPropertyNamePolicyAttribute PropertyNamePolicyDefault { get; set; }
-    }
-
-    // Each JsonPropertyInfo is an instance of this (to avoid boxing)
-    public class JsonPropertyInfo<TValue> : JsonPropertyInfo
-    {
-        // This is the effective data type converter that will be used.
-        public JsonDataTypeConverter<TValue> DataTypeConverter { get; }
-    }
-}
-```
-
-Note that `JsonClassInfo`, `JsonPropertyInfo`, and `JsonPropertyInfo<TValue>` are all used internally by the (de)serializer for its core implementation and thus are not created just for extensibility.
+## Extensibility and manual (de)serialization
+This is a pending feature and design work and requirement gathering is underway. There will likely be two types of converters: a "value" converter for primitive types that have a single value in JSON (such as a number or string) and an "object" converter that allows access to the reader and writer and will have before- and after- callbacks. These converters will be registered with `JsonSerializerOptions`.
 
 # API comparison to Json.NET
 ## Simple scenario
 Json.NET:
 ```cs
     Person person = ...;
-    string result = JsonConvert.SerializeObject(person);
-    person = JsonConvert.DeserializeObject(person);
+    string json = JsonConvert.SerializeObject(person);
```

Comment:
It would be really useful to flesh these samples out a bit more and add to the porting guide: https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/porting_guide/README.md#guide-to-porting-from-newtonsoftjson-to-systemtextjson